app/models/post.rb

```diff
@@ -1,4 +1,14 @@
 class Post < ApplicationRecord
-  belongs_to :user
-  has_many :comments, :likes
+  belongs_to :author, class_name: :User
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+
+  def increase_post_counter
+    user.posts_counter += 1
+    user.save
+  end
```

Comment:
- [ ] The post belongs to an `author`, not `user`. So we need to change `user` here to `author`.

- **[OPTIONAL]** I think it would be nice if you triggered this method using the `after_save` active record callback so that when a post is saved Rails automatically calls this function. I also think we can refactor this code to be more brief.

  ```
  .
  .
  after_save :increase_post_counter
  .
  .
  private

  def increase_post_counter
    author.increment!(:posts_counter)
  end
  ```